packages/components/src/components/form/bal-time-input/bal-time-input.tsx

```diff
@@ -0,0 +1,318 @@
+   */
+  @Prop() disabled = false
+
+  /**
+   * If `true` the element can not mutated, meaning the user can not edit the control.
+   */
+  @Prop() readonly = false
+
+  /**
+   * Set the amount of time, in milliseconds, to wait to trigger the `balChange` event after each keystroke. This also impacts form bindings such as `ngModel` or `v-model`.
+   */
+  @Prop() debounce = 0
+
+  @Watch('debounce')
+  protected debounceChanged() {
+    this.balChange = debounceEvent(this.balChange, this.debounce)
+  }
+
+  /**
+   * The value of the input.
+   */
+  @Prop({ mutable: true }) value?: string = undefined
+
+  /**
+   * Emitted when a keyboard input occurred.
+   */
+  @Event() balInput!: EventEmitter<Events.BalInputTimeInputDetail>
+
+  /**
+   * Emitted when the value has changed.
+   */
+  @Event() balChange!: EventEmitter<Events.BalInputTimeChangeDetail>
+
+  /**
+   * Emitted when the input loses focus.
+   */
+  @Event() balBlur!: EventEmitter<FocusEvent>
+
+  /**
+   * Emitted when the input has focus.
+   */
+  @Event() balFocus!: EventEmitter<FocusEvent>
+
+  /**
+   * Emitted when a keyboard key has pressed.
+   */
+  @Event() balKeyPress!: EventEmitter<KeyboardEvent>
+
+  /**
+   * Emitted when the input has clicked.
+   */
+  @Event() balClick!: EventEmitter<MouseEvent>
+
+  @Listen('click', { capture: true, target: 'document' })
+  listenOnClick(event: UIEvent) {
+    inputListenOnClick(this, event)
+  }
+
+  private resetHandlerTimer?: NodeJS.Timer
+
+  @Listen('reset', { capture: true, target: 'document' })
+  resetHandler(event: UIEvent) {
+    const formElement = event.target as HTMLElement
+    if (formElement?.contains(this.el)) {
+      inputHandleReset(this, this.initialValue, this.resetHandlerTimer)
+    }
+  }
+
+  connectedCallback() {
+    this.debounceChanged()
+    attachComponentToConfig(this)
+    this.initialValue = this.value || ''
+  }
+
+  componentDidLoad() {
+    this.inputValue = this.value
+  }
+
+  componentWillLoad() {
+    this.inheritedAttributes = inheritAttributes(this.el, ['aria-label', 'tabindex', 'title'])
+  }
+
+  disconnectedCallback() {
+    detachComponentToConfig(this)
+  }
+
+  configChanged(state: BalConfigState): void {
+    this.language = state.language
+    this.region = state.region
+
+    if (!this.hasFocus && this.nativeInput) {
+      this.nativeInput.value = this.getFormattedValue()
+    }
+  }
+
+  /**
+   * Sets focus on the native `input`. Use this method instead of the global
+   * `input.focus()`.
+   */
+  @Method()
+  async setFocus() {
+    inputSetFocus(this)
+  }
+
+  /**
+   * Sets blur on the native `input`. Use this method instead of the global
+   * `input.blur()`.
+   * @internal
+   */
+  @Method()
+  async setBlur() {
+    inputSetBlur(this)
+  }
+
+  /**
+   * Returns the native `<input>` element used under the hood.
+   */
+  @Method()
+  getInputElement(): Promise<HTMLInputElement> {
```

Comment:
- [ ] just put async in front of it the you can get rid off `Promise.resolve` and just return the value